server/src/test/java/org/elasticsearch/action/admin/indices/settings/get/TransportGetSettingsActionIT.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.settings.get;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESIntegTestCase;
+
+public class TransportGetSettingsActionIT extends ESIntegTestCase {
```

Comment:
this is ok to have, but I meant a unit test while this one is more of an integration test as it requires to send requests to an Elasticsearch cluster. I would call it `GetSettingsActionIT` instead. Do you think we could have the same as a unit test, maybe as a followup?

Reply:
If you look at, for example, `TransportBulkActionTests`, that will give you an idea how to construct a transport action for tests.

Reply:
`this.settings` is node-level settings from startup; this will never have index-scoped settings in it.

Reply:
I think you could pass in `Settings.EMPTY` when constructing this for the purpose of tests too and it will be fine.

Reply:
Thanks @jasontedor that is making this make a bunch more sense